### DISCUSSION TITLE
Question regarding actions

### DISCUSSION THREAD

**Discussion by ScottHoang** [ds4dm/ml4co-competition#45] [Q&A] [answered]
for primal task's item placement, I see that action contains two variables of 1050 sizes.
actions is : [([614, 215, 563, ..., 588, 342, 246], dtype=uint64), array([0, 1, 1, ..., 0, 1, 0]))
what are they?
is action[0] indicated int solution and action[1] indicated feasibility?
thank you!

**Comment by gasse** [comment, marked as answer]
Hi @ScottHoang ,

Actions in the primal environment are partial primal solutions, i.e., value assignments for a subset of the problem's variables. In Ecole we chose to implement such an object following a [coordinate list (COO)](https://en.wikipedia.org/wiki/Sparse_matrix#Coordinate_list_(COO)) format. The first element is a list of indices, which indicate the variables that are affected a value to (variable indices in SCIP's transformed problem), while the second element is a list of the same size indicating the values affected to those variables (scalars). Primal actions require that only integer variables get assigned a value (not necessarily all integer variables though), while values for the rest of the variables are assigned by solving an LP relaxation after the assigned integer variables are fixed to their values.

Hope that helps,

Best,
Maxime
